job_board/portals/wellfound.py

```diff
@@ -30,7 +54,16 @@ def get_jobs(self) -> list[Job]:
         jobs_data = []
         while True:
             url = f"{self.url}?page={page_number}"
-            content = make_scrapfly_request(url=url, asp=True)
+            for attempt in Retrying(
+                stop=stop_after_attempt(5),
+                retry=retry_if_exception(is_retryable),
+                wait=wait_exponential(min=1, multiplier=1, max=5),
+                reraise=True,
+                before_sleep=before_sleep_logging,
+            ):
+                with attempt:
+                    content = make_scrapfly_request(url=url, asp=True)
+
```

Comment:
Can we convert it to decorator?

Right now, let's say we have to call make_scrapefly_request from some other place, we have to write this retry logic again there..


I'm thinking of something like

```
@retry(attempts=5, ...)
def make_scrapefly_req(...)
```


And then use this function wherever you want..